tap_google_analytics/client.py

```diff
@@ -13,12 +13,29 @@
 
 LOGGER = singer.get_logger()
 
+def is_retryable_403(e):
+    response = e.response
+    if not _is_json(response):
+        return False
+
+    retryable_errors = ["userRateLimitExceeded", "rateLimitExceeded", "quotaExceeded"]
+    error_reasons = [error.get('reason') for error in response.json().get('errors',[])]
+    for retryable_error in retryable_errors:
+        if retryable_error in error_reasons:
+            return True
+    return False
+
 def should_giveup(e):
-    if e.response.status_code == 429:
+    should_retry = e.response.status_code == 429 or is_retryable_403(e)
```

Comment:
Can you use a retry library instead of doing all this custom logic? There's some nice retry libraries in python where you can just annotate a method I believe. This is the one I'm familiar with: https://github.com/jd/tenacity

Reply:
Oh nevermind the retrying is part of the client library...